### ISSUE TITLE
Results of ConvBwdWeightsAlgoDirect and ConvBwdWeightsAlgoGEMM mismatch for some specific parameters

### ISSUE THREAD

**Issue by xiaocenxiaocen** [ROCmSoftwarePlatform/MIOpen#77]
Hey guys, I am testing the correctness of miopen's ConvolutionBackwardWeights operator. And I am having the following issue.

**System Information**
 

- Have I written custom code: Yes
   I tried to run the following code:
```C++
#define __HIP_PLATFORM_HCC__

#include <miopen/miopen.h>
#include <rocblas.h>

#include <assert.h>
#include <hip/hip_runtime.h>
#include <hip/hip_runtime_api.h>
#include <string>
#include <vector>

#define opencl_check(x) (x)

#define hip_check(x) (x)

class ROCMTimer {
   private:
    bool m_started, m_stopped;
    hipEvent_t m_event_start, m_event_end;
    hipStream_t stream;

   public:
    ROCMTimer(hipStream_t strm) : stream{strm} {
        hip_check(hipEventCreate(&m_event_start));
        hip_check(hipEventCreate(&m_event_end));
        reset();
    }
    ~ROCMTimer() {
        hip_check(hipEventDestroy(m_event_start));
        hip_check(hipEventDestroy(m_event_end));
    }
    void start() {
        megdnn_assert(!m_started);
        megdnn_assert(!m_stopped);

        m_started = true;
        hip_check(hipEventRecord(m_event_start, stream));
    }
    void stop() {
        megdnn_assert(m_started);
        megdnn_assert(!m_stopped);
        m_stopped = true;
        hip_check(hipEventRecord(m_event_end, stream));
    }
    float get_time_in_ms() const {
        megdnn_assert(m_started);
        megdnn_assert(m_stopped);
        hip_check(hipEventSynchronize(m_event_end));
        float ms;
        hip_check(hipEventElapsedTime(&ms, m_event_start, m_event_end));
        return ms;
    }
    void reset() {
        m_started = false;
        m_stopped = false;
    }
};

void init_devices() {
    int devcount;
    hipGetDeviceCount(&devcount);
    hipSetDevice(0);
}

float check_err(float* lhs, float* rhs, size_t size) {
    float error = 0.f;
    for (size_t i = 0; i < size; i++) {
        auto vlhs = lhs[i];
        auto vrhs = rhs[i];
        auto vmax = std::max(std::abs(vlhs), std::abs(vrhs));
        auto verr = std::abs(vlhs - vrhs) / vmax;
        if (verr > error) {
            error = verr;
        }
        if (verr > 1e-3) {
            printf("error = %.7f, vlhs = %.7f, vrhs = %.7f, index = %zu\n",
                   verr, vlhs, vrhs, i);
        }
    }
}

void conv_bwd_wrw(float* y, float* x, float* w, int N, int OC, int IC, int H,
                  int W, int pad_h, int pad_w, int kh, int kw, int sh, int sw) {
    int OH = (H + 2 * pad_h - kh) / sh + 1;
    int OW = (W + 2 * pad_w - kw) / sw + 1;

    for (size_t n = 0; n < N; ++n) {
        for (size_t oc = 0; oc < OC; ++oc)
            for (size_t oh = 0; oh < OH; ++oh)
                for (size_t ow = 0; ow < OW; ++ow) {
                    float& dval =
                        y[n * OC * OH * OW + oc * OH * OW + oh * OW + ow];

                    for (size_t fh = 0; fh < kh; ++fh)
                        for (size_t fw = 0; fw < kw; ++fw) {
                            size_t ih = sh * oh + fh - pad_h,
                                   iw = sw * ow + fw - pad_w;
                            // here ih and iw are represented in unsigned int
                            // they will become very large if underflow occurs
                            if (ih < H && iw < W) {
                                size_t ic0 = 0, ic1 = ic0 + IC;
                                for (size_t ic = ic0; ic < ic1; ++ic) {
                                    float& sval = x[n * IC * H * W +
                                                    ic * H * W + ih * W + iw];
                                    float& wval =
                                        w[oc * IC * kh * kw + ic * kh * kw +
                                          fh * kw + fw];
                                    wval += sval * dval;
                                }
                            }
                        }
                }
    }
}

int main() {
    int N = 2;
    int OC = 1;
    int IC = 2;
    int H = 50;
    int W = 55;
    int pad_h = 0, pad_w = 0;
    int kh = 3, kw = 3;
    int sh = 1, sw = 2;
    int OH = (H + 2 * pad_h - kh) / sh + 1;
    int OW = (W + 2 * pad_w - kw) / sw + 1;

    hipSetDevice(0);
    hipStream_t strm;
    hipStreamCreateWithFlags(&strm, hipStreamNonBlocking);
    printf("%p\n", strm);

    miopenConvolutionMode_t c_mode = miopenConvolution;
    miopenHandle_t handle;
    miopenStatus_t err = miopenCreate(&handle);

    // miopenStatus_t err = miopenCreateWithStream(&handle, strm);
    // miopenSetStream(handle, strm);

    //    rocblas_handle hdl;
    //    rocblas_create_handle(&hdl);
    //    rocblas_set_stream(hdl, strm);
    //    printf("%p\n", strm);
    miopenTensorDescriptor_t x_desc, w_desc, y_desc;
#define init_desc(desc, n, c, h, w)                                       \
    do {                                                                  \
        err = miopenCreateTensorDescriptor(&desc);                        \
        err = miopenSet4dTensorDescriptor(desc, miopenFloat, n, c, h, w); \
    } while (0)
    init_desc(x_desc, N, IC, H, W);
    init_desc(w_desc, OC, IC, kh, kw);
    init_desc(y_desc, N, OC, OH, OW);

    init_devices();
    float *ts_x, *ts_w, *ts_y, *ws;
    hipMalloc(&ts_x, N * IC * H * W * sizeof(float));
    hipMalloc(&ts_w, OC * IC * kh * kw * sizeof(float));
    hipMalloc(&ts_y, N * OC * OH * OW * sizeof(float));

    float *h_ts_x, *h_ts_w_direct, *h_ts_w_gemm, *h_ts_y, *h_ts_w;
    h_ts_x = (float*)malloc(sizeof(float) * N * IC * H * W);
    h_ts_w_direct = (float*)malloc(sizeof(float) * OC * IC * kh * kw);
    h_ts_w_gemm = (float*)malloc(sizeof(float) * OC * IC * kh * kw);
    h_ts_w = (float*)malloc(sizeof(float) * OC * IC * kh * kw);
    h_ts_y = (float*)malloc(sizeof(float) * N * OC * OH * OW);

    srand((unsigned int)(time(NULL)));
    for (size_t i = 0; i < N * IC * H * W; i++) {
        h_ts_x[i] = 1.f * rand() / RAND_MAX;
    }

    for (size_t i = 0; i < N * OC * OH * OW; i++) {
        h_ts_y[i] = 1.f * rand() / RAND_MAX;
    }
    for (size_t i = 0; i < OC * IC * kh * kw; i++) {
        h_ts_w[i] = 1.f * rand() / RAND_MAX;
    }
    hipMemcpyAsync(ts_x, h_ts_x, sizeof(float) * N * IC * H * W,
                   hipMemcpyHostToDevice, strm);
    hipMemcpyAsync(ts_y, h_ts_y, sizeof(float) * N * OC * OH * OW,
                   hipMemcpyHostToDevice, strm);
    //    hipMemcpyAsync(ts_w, h_ts_w, sizeof(float) * OC * IC * kh * kw,
    //                   hipMemcpyHostToDevice, strm);

    ROCMTimer timer{strm};
    timer.reset();
    miopenConvolutionDescriptor_t conv_desc;
    err = miopenCreateConvolutionDescriptor(&conv_desc);
    err = miopenInitConvolutionDescriptor(conv_desc, c_mode, pad_h, pad_w, sh,
                                          sw, 1, 1);

    size_t ws_size;
    err = miopenConvolutionBackwardWeightsGetWorkSpaceSize(
        handle, y_desc, x_desc, conv_desc, w_desc, &ws_size);
    hipMalloc(&ws, ws_size);

    int ret_algo_count;
    int req_algo_count = 10;
    miopenConvAlgoPerf_t algo_perf[req_algo_count];
    printf(
        "inp={%d,%d,%d,%d} filter={%d,%d,%d,%d} dst={%d,%d,%d,%d} pad=%dx%d, "
        "stride=%dx%d\n",
        N, IC, H, W, OC, IC, kh, kw, N, OC, OH, OW, pad_h, pad_w, sh, sw);
    err = miopenFindConvolutionBackwardWeightsAlgorithm(
        handle, y_desc, ts_y, x_desc, ts_x, conv_desc, w_desc, ts_w,
        req_algo_count, &ret_algo_count, algo_perf, ws, ws_size, false);
    for (int i = 0; i < ret_algo_count; i++) {
        printf("INFO: conv fwd algo = %d\n", algo_perf[i].bwd_weights_algo);
        printf("INFO: time = %f\n", algo_perf[i].time);
        printf("INFO: memory = %zu\n", algo_perf[i].memory);
    }

    {
        timer.start();
        float alpha = 1.0;
        float beta = 0.0;
        miopenConvBwdWeightsAlgorithm_t algo = algo_perf[0].bwd_weights_algo;
        algo = miopenConvolutionBwdWeightsAlgoGEMM;
        hipMemset(ts_w, 0, sizeof(float) * OC * IC * kh * kw);
        err = miopenConvolutionBackwardWeights(
            handle, &alpha, y_desc, ts_y, x_desc, ts_x, conv_desc, algo, &beta,
            w_desc, ts_w, ws, ws_size);
        hipMemcpy(h_ts_w_gemm, ts_w, sizeof(float) * OC * IC * kh * kw,
                  hipMemcpyDeviceToHost);
        timer.stop();
    }

    {
        timer.start();
        float alpha = 1.0;
        float beta = 0.0;
        miopenConvBwdWeightsAlgorithm_t algo = algo_perf[0].bwd_weights_algo;
        algo = miopenConvolutionBwdWeightsAlgoDirect;
        hipMemset(ts_w, 0, sizeof(float) * OC * IC * kh * kw);
        err = miopenConvolutionBackwardWeights(
            handle, &alpha, y_desc, ts_y, x_desc, ts_x, conv_desc, algo, &beta,
            w_desc, ts_w, ws, ws_size);
        hipMemcpy(h_ts_w_direct, ts_w, sizeof(float) * OC * IC * kh * kw,
                  hipMemcpyDeviceToHost);
        timer.stop();
    }

    auto error = check_err(h_ts_w_direct, h_ts_w_gemm, OC * IC * kh * kw);
    memset(h_ts_w, 0, sizeof(float) * OC * IC * kh * kw);
    conv_bwd_wrw(h_ts_y, h_ts_x, h_ts_w, N, OC, IC, H, W, pad_h, pad_w, kh, kw,
                 sh, sw);
    error = check_err(h_ts_w, h_ts_w_gemm, OC * IC * kh * kw);
    printf("%.4f\n", error);
    error = check_err(h_ts_w, h_ts_w_direct, OC * IC * kh * kw);
    printf("%.4f\n", error);

    hipFree(ts_x);
    hipFree(ts_w);
    hipFree(ts_y);
    hipFree(ws);

    free(h_ts_x);
    free(h_ts_w_direct);
    free(h_ts_w_gemm);
    free(h_ts_y);

    miopenDestroyTensorDescriptor(x_desc);
    miopenDestroyTensorDescriptor(w_desc);
    miopenDestroyTensorDescriptor(y_desc);
    miopenDestroyConvolutionDescriptor(conv_desc);
    err = miopenDestroy(handle);
}
```

- Results of hipconfig --full
```
HIP version  : 1.5.18494

== hipconfig
HIP_PATH     : /opt/rocm
HIP_PLATFORM : hcc
CPP_CONFIG   :  -D__HIP_PLATFORM_HCC__=   -I/opt/rocm/include -I/opt/rocm/hcc/include

== hcc
HSA_PATH     : /opt/rocm/hsa
HCC_HOME     : /opt/rocm/hcc
HCC clang version 8.0.0 (ssh://gerritgit/compute/ec/hcc-tot/clang 6ec3c61e09fbb60373eaf5a40021eb862363ba2c) (ssh://gerritgit/lightning/ec/llvm ab3b88ffc2ae50f55361a49aec89f6e95d9d0ec4) (based on HCC 1.3.18482-757fb49-6ec3c61-ab3b88f )
Target: x86_64-unknown-linux-gnu
Thread model: posix
InstalledDir: /opt/rocm/hcc/bin
LLVM (http://llvm.org/):
  LLVM version 8.0.0svn
  Optimized build.
  Default target: x86_64-unknown-linux-gnu
  Host CPU: haswell

  Registered Targets:
    amdgcn - AMD GCN GPUs
    r600   - AMD GPUs HD2XXX-HD6XXX
    x86    - 32-bit X86: Pentium-Pro and above
    x86-64 - 64-bit X86: EM64T and AMD64
HCC-cxxflags :  -hc -std=c++amp -I/opt/rocm/hcc/include -I/opt/rocm/includeHCC-ldflags  :  -hc -std=c++amp -L/opt/rocm/hcc/lib -Wl,--rpath=/opt/rocm/hcc/lib -ldl -lm -lpthread -lhc_am -Wl,--whole-archive -lmcwamp -Wl,--no-whole-archive

=== Environment Variables
PATH=/opt/rocm/bin:/opt/rocm/hcc/bin:/opt/rocm/hip/bin:/opt/rocm/hsa/bin:/opt/rocm/miopen/bin:/opt/rocm/miopengemm/bin:/opt/rocm/rocblas/bin
LD_LIBRARY_PATH=/opt/rocm/lib:/opt/rocm/hcc/lib:/opt/rocm/hip/lib:/opt/rocm/hsa/lib:/opt/rocm/miopen/lib:/opt/rocm/miopengemm/lib:/opt/rocm/rocblas/lib:/opt/rocm/opencl/lib/x86_64:/opt/rocm/rocrand/lib:

== Linux Kernel
Linux 4.13.16-041316-generic #201711240901 SMP Fri Nov 24 09:02:42 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 16.04.5 LTS
Release:	16.04
Codename:	xenial
```

- Version of MIOpen: 1.7.0

**Describe of current Behavior**
The results of ConvBwdWeightsAlgoDirect and ConvBwdWeightsAlgoGEMM mismatch. Result of ConvBwdWeightsAlgoGEMM is identical to the naive implementation. But result of ConvBwdWeightsAlgoDirect is different with the result of naive implementation.

**Describe of excepted Behavior**
Results of ConvBwdWeightsAlgoDirect and ConvBwdWeightsAlgoGEMM are same with each other for the specific parameters.

**Code to Reproduce the issue**
Use the following command to compile the code above
```
hipcc -Wall -std=c++14 test.cpp -L/opt/rocm/lib -lMIOpen -lhip_hcc -lOpenCL -lmiopengemm -lrocblas
```
And then run the program with 
```
./a.out
```
**Other Info/logs**
- MIOpen Backend: HIP

**Comment by whchung** [comment]
@xiaocenxiaocen thanks for providing this reduced case. just to be sure we are on the same page, could you help provide the outputs on your side so we can be sure we are getting the same results with you?

**Comment by whchung** [comment]
@xiaocenxiaocen also could you provide the AMD GPU hardware you are using? is it a Polaris, or a Vega?

**Comment by xiaocenxiaocen** [comment]
@whchung Vega(gfx900)

**Comment by xiaocenxiaocen** [comment]
@whchung 
my output
-> % ./a.out
```
0x1077670
inp={2,2,50,55} filter={1,2,3,3} dst={2,1,48,27} pad=0x0, stride=1x2
INFO: conv fwd algo = 1
INFO: time = 0.043409
INFO: memory = 144
INFO: perf = 2.149600 GFLOPS
INFO: conv fwd algo = 0
INFO: time = 0.071706
INFO: memory = 93312
INFO: perf = 1.301314 GFLOPS
error = 0.4825625, vlhs = 339.2410889, vrhs = 655.6175537, index = 0
error = 0.4831659, vlhs = 335.6265259, vrhs = 649.3892822, index = 1
error = 0.4855363, vlhs = 331.5067139, vrhs = 644.3734131, index = 2
error = 0.4722224, vlhs = 339.5617676, vrhs = 643.3803711, index = 3
error = 0.4699603, vlhs = 339.1345825, vrhs = 639.8286133, index = 4
error = 0.4802689, vlhs = 336.0886841, vrhs = 646.6588135, index = 5
error = 0.4978431, vlhs = 319.2985229, vrhs = 635.8540649, index = 6
error = 0.5120595, vlhs = 315.8125610, vrhs = 647.2357178, index = 7
error = 0.5108531, vlhs = 314.2026978, vrhs = 642.3483276, index = 8
error = 0.4751219, vlhs = 344.0303955, vrhs = 655.4481201, index = 9
error = 0.4920473, vlhs = 336.1373291, vrhs = 661.7493286, index = 10
error = 0.4803235, vlhs = 339.0942383, vrhs = 652.5101929, index = 11
error = 0.4831491, vlhs = 334.4456177, vrhs = 647.0833130, index = 12
error = 0.4842871, vlhs = 336.2611084, vrhs = 652.0315552, index = 13
error = 0.4836295, vlhs = 337.1076050, vrhs = 652.8405762, index = 14
error = 0.4930795, vlhs = 328.9961243, vrhs = 649.0093384, index = 15
error = 0.5038261, vlhs = 326.1661377, vrhs = 657.3625488, index = 16
error = 0.4951567, vlhs = 326.3660889, vrhs = 646.4700317, index = 17
0.0000
error = 0.4825634, vlhs = 655.6186523, vrhs = 339.2410889, index = 0
error = 0.4831657, vlhs = 649.3889771, vrhs = 335.6265259, index = 1
error = 0.4855371, vlhs = 644.3744507, vrhs = 331.5067139, index = 2
error = 0.4722218, vlhs = 643.3796997, vrhs = 339.5617676, index = 3
error = 0.4699595, vlhs = 639.8276978, vrhs = 339.1345825, index = 4
error = 0.4802694, vlhs = 646.6594238, vrhs = 336.0886841, index = 5
error = 0.4978429, vlhs = 635.8538208, vrhs = 319.2985229, index = 6
error = 0.5120597, vlhs = 647.2360229, vrhs = 315.8125610, index = 7
error = 0.5108527, vlhs = 642.3478394, vrhs = 314.2026978, index = 8
error = 0.4751220, vlhs = 655.4483643, vrhs = 344.0303955, index = 9
error = 0.4920477, vlhs = 661.7497559, vrhs = 336.1373291, index = 10
error = 0.4803234, vlhs = 652.5100708, vrhs = 339.0942383, index = 11
error = 0.4831491, vlhs = 647.0833130, vrhs = 334.4456177, index = 12
error = 0.4842878, vlhs = 652.0324707, vrhs = 336.2611084, index = 13
error = 0.4836296, vlhs = 652.8406372, vrhs = 337.1076050, index = 14
error = 0.4930793, vlhs = 649.0090942, vrhs = 328.9961243, index = 15
error = 0.5038267, vlhs = 657.3634033, vrhs = 326.1661377, index = 16
error = 0.4951569, vlhs = 646.4702759, vrhs = 326.3660889, index = 17
326.3661
```

**Comment by atamazov** [comment]
This is normal when different algorithms have different precision. The requirement is that implementation/algorithm shall have _good enough_ precision. 

Precision also depend on input data (this is especially so for Winograd algorithm). I recommend using random distributions similar to real-life ones. For data this means something positive, in range [0..1], I think.

However, in this case our implementation of Direct WrW algorithm fails validation:
```
$ ./bin/MIOpenDriver conv -n 2 -k 1 -c 2 -H 50 -W 55 -p 0 -q 0 -x 3 -y 3 -u 1 -v 2 -t 1 -w 1 -V 1
...
MIOpen Forward Conv. Algorithm: 0
...
Forward Convolution Verifies on CPU and GPU (2.39599e-08)
...
MIOpen Backward Data Conv. Algorithm: 0
GPU Kernel Time Backward Data Conv. Elapsed: 0.032907 ms (average)
...
MIOpen Backward Weights Conv. Algorithm: 1
...
Backward Convolution Data Verifies on CPU and GPU (1.88443e-08)
...
Backward Convolution Weights Failed: 0.479145 <<<---
```

GEMM implementation of WrW passes validation.

@xiaocenxiaocen Please confirm that the following driver parameters are match to the convolution config you are using: `-n 2 -k 1 -c 2 -H 50 -W 55 -p 0 -q 0 -x 3 -y 3 -u 1 -v 2 `

We are going to fix this in next release.

**Comment by xiaocenxiaocen** [comment]
> @xiaocenxiaocen Please let us know how urgent/important is this issue for you.

When will the next version release?

**Comment by atamazov** [comment]
@xiaocenxiaocen The attached patch shall resolve this problem for you until release with full-blown fix (either 1.7.1 or 1.8) is arrived.

[pr1473.diff.txt](https://github.com/ROCmSoftwarePlatform/MIOpen/files/2821863/pr1473.diff.txt)

**Comment by xiaocenxiaocen** [comment]
Do you have any progress?The MIOpen(version 1.7.1) has been released. And the release log claimed this bug had been fixed. But I tried the following command
```
MIOpenDriver: conv -n 2 -k 1 -c 2 -H 50 -W 55 -p 0 -q 0 -x 3 -y 3 -u 1 -v 2 -t 1 -w 1 -V 1
MIOpen(HIP): Info [DetectAmdRocmMetadataVersion] AMDHSA_1_0
MIOpen(HIP): Info [FindSolutionImpl] ConvOclBwdWrW2 (not searchable)
MIOpen(HIP): Info [SetValues] 2-50-55-3x3-1-48-27-2-0x0-2x1-1x1-0-NCHW-FP32-F, content inserted: miopenConvolutionFwdAlgoGEMM:gemm,0.033482,93312,0_0_1296_18_1296_1296_1_18
MIOpen(HIP): Info [FindConvFwdAlgorithm] miopenConvolutionFwdAlgoGEMM   0.033482        93312
Wall-clock Time Forward Conv. Elapsed: 0.218282 ms
MIOpen Forward Conv. Algorithm: 0
GPU Kernel Time Forward Conv. Elapsed: 0.029811 ms (average)
Forward Convolution Verifies on CPU and GPU (4.46837e-08)
MIOpen(HIP): Info [FindConvBwdDataAlgorithm] miopenConvolutionBwdDataAlgoGEMM   0.038814        93312
Wall-clock Time Backward Data Conv. Elapsed: 0.203102 ms
MIOpen Backward Data Conv. Algorithm: 0
GPU Kernel Time Backward Data Conv. Elapsed: 0.028494 ms (average)
MIOpen(HIP): Info [FindSolutionImpl] ConvOclBwdWrW2 (not searchable)
MIOpen(HIP): Info [FindConvBwdWeightsAlgorithm] ConvOclBwdWrW2: MIOpenCvBwdWrW/MIOpenCvBwdWrW_rdc: 0.03926 < 3.40282e+38, workspce_sz = 144
MIOpen(HIP): Info [FindConvBwdWeightsAlgorithm] Selected: ConvOclBwdWrW2: MIOpenCvBwdWrW/MIOpenCvBwdWrW_rdc: 0.03926, workspce_sz = 144
MIOpen(HIP): Info [FindConvBwdWeightsAlgorithm] miopenConvolutionBwdWeightsAlgoDirect   0.03926 144
MIOpen(HIP): Info [FindConvBwdWeightsAlgorithm] miopenConvolutionBwdWeightsAlgoGEMM     0.0483  93312
Wall-clock Time Backward Weights Conv. Elapsed: 0.134378 ms
MIOpen Backward Weights Conv. Algorithm: 1
GPU Kernel Time Backward Weights Conv. Elapsed: 0.030964 ms (average)
Backward Convolution Data Verifies on CPU and GPU (1.50604e-08)
Backward Convolution Weights Failed: 0.477952
```
The MIOpenDriver still failed.

**Comment by atamazov** [comment]
Please check that kernel cache is wiped out.

/cc @daniellowell

**Comment by xiaocenxiaocen** [comment]
> Please check that kernel cache is wiped out.
> 
> /cc @daniellowell

how can I wipe out the kernel cache? I have deleted the directory ~/.cache/miopen/1.7.1.

**Comment by atamazov** [comment]
@xiaocenxiaocen 
> how can I wipe out the kernel cache? I have deleted the directory ~/.cache/miopen/1.7.1.

That's right way. AFAIR you can also set `MIOPEN_DISABLE_CACHE=1` in the env to disable cache temporarily.

I am looking into the issue right now.

**Comment by atamazov** [comment]
@xiaocenxiaocen 

> Do you have any progress?

The full-blown fix will be included into 1.8 release.

> The MIOpen(version 1.7.1) has been released. And the release log claimed this bug had been fixed.

This record in the log is incorrect. Please apply the [diff provided above](https://github.com/ROCmSoftwarePlatform/MIOpen/issues/77#issuecomment-459733235) to have the issue fixed in 1.7.1.

Sorry for confusion.

**Comment by daniellowell** [comment]
Fixed.
